### PR TITLE
SASS-6339: Submit expenses tailoring answers to backend

### PR DESCRIPTION
**Pull request by lukastymo**

### CODE REVIEW

File: app/controllers/journeys/expenses/goodsToSellOrUse/GoodsToSellOrUseCYAController.scala [hunk 1]
```diff
@@ -61,16 +63,13 @@ class GoodsToSellOrUseCYAController @Inject() (override val messagesApi: Message
 
   def onSubmit(taxYear: TaxYear, businessId: BusinessId): Action[AnyContent] = (identify andThen getData andThen requireData).async {
     implicit request =>
-      val journeyAnswers = (request.userAnswers.data \ businessId.value).as[GoodsToSellOrUseJourneyAnswers]
-      val context        = SubmissionContext(taxYear, Nino(request.user.nino), businessId, Mtditid(request.user.mtditid), ExpensesGoodsToSellOrUse)
+      val nextRoute = navigator.nextNormalRoute(GoodsToSellOrUseCYAPage, request.userAnswers, taxYear, businessId, accountingType = None).url
+      val result = service
+        .submitAnswers[GoodsToSellOrUseJourneyAnswers](taxYear, businessId, request.mtditid, ExpensesGoodsToSellOrUse, request.userAnswers)
```
Comment (harry-ritchie) [inline app/controllers/journeys/expenses/goodsToSellOrUse/GoodsToSellOrUseCYAController.scala:68]: Can we use `SubmissionContext` for param holder?

File: app/connectors/SelfEmploymentConnector.scala [hunk 2]
```diff
@@ -90,6 +98,11 @@ class SelfEmploymentConnector @Inject() (http: HttpClient, appConfig: FrontendAp
     )
   }
 
-  private def buildUrl(url: String) = appConfig.selfEmploymentBEBaseUrl + url
-
+  def submitAnswers[A: Writes](taxYear: TaxYear, businessId: BusinessId, mtditid: Mtditid, journey: Journey, answers: A)(implicit
```
[the remark this answers is not in the document]
Comment (lukastymo) [reply app/connectors/SelfEmploymentConnector.scala:101]: sure

File: app/connectors/package.scala [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+import connectors.httpParser.JourneyStateParser.pagerDutyError
+import models.common.Mtditid
+import models.errors.HttpError
+import play.api.libs.json.Writes
+import uk.gov.hmrc.http.{HeaderCarrier, HttpClient, HttpReads, HttpResponse}
+
+import scala.concurrent.{ExecutionContext, Future}
+
+package object connectors {
+  type ConnectorNoResult = Either[HttpError, Unit]
```
[the remark this answers is not in the document]
Comment (lukastymo) [reply app/connectors/package.scala:27]: done

File: app/controllers/journeys/expenses/goodsToSellOrUse/GoodsToSellOrUseCYAController.scala [hunk 4]
```diff
@@ -61,16 +63,13 @@ class GoodsToSellOrUseCYAController @Inject() (override val messagesApi: Message
 
   def onSubmit(taxYear: TaxYear, businessId: BusinessId): Action[AnyContent] = (identify andThen getData andThen requireData).async {
     implicit request =>
-      val journeyAnswers = (request.userAnswers.data \ businessId.value).as[GoodsToSellOrUseJourneyAnswers]
-      val context        = SubmissionContext(taxYear, Nino(request.user.nino), businessId, Mtditid(request.user.mtditid), ExpensesGoodsToSellOrUse)
+      val nextRoute = navigator.nextNormalRoute(GoodsToSellOrUseCYAPage, request.userAnswers, taxYear, businessId, accountingType = None).url
```
[the remark this answers is not in the document]
Comment (lukastymo) [reply app/controllers/journeys/expenses/goodsToSellOrUse/GoodsToSellOrUseCYAController.scala:66]: Other CYAs page use navigator.

I think it make sense to move the redirection concern to one class, so the navigator always decides where to redirect, not the Controller itself

File: app/controllers/journeys/expenses/goodsToSellOrUse/GoodsToSellOrUseCYAController.scala (hunk 1 above)
Comment (lukastymo) [reply app/controllers/journeys/expenses/goodsToSellOrUse/GoodsToSellOrUseCYAController.scala:68]: done

File: app/services/SelfEmploymentService.scala [hunk 5]
```diff
@@ -73,6 +75,12 @@ class SelfEmploymentService @Inject() (connector: SelfEmploymentConnector, sessi
       updatedAnswers <- Future.fromTry(userAnswers.set[A](page, value, Some(businessId)))
       _              <- sessionRepository.set(updatedAnswers)
     } yield updatedAnswers
+
+  def submitAnswers[SubsetOfAnswers: Format](taxYear: TaxYear, businessId: BusinessId, mtditid: Mtditid, journey: Journey, userAnswers: UserAnswers)(
```
[the remark this answers is not in the document]
Comment (lukastymo) [reply app/services/SelfEmploymentService.scala]: I added happy path test, 

but I don't see any value in  unhappy path test as it just call the connectors without doing anything (if excpetion is thrown it is passed downstream).

**Review by harry-ritchie — APPROVED** [review]

**Pull request merged**